backend/apps/medical_catalog/services/medicine_service.py

```
@@ -10,29 +13,70 @@ def create(*, data: dict, user=None) -> Medicine:
-        return instance
+        try:
+            instance = Medicine(**filtered_data)
+            instance.full_clean()
+            instance.save()
+            return instance
+        except ValidationError as e:
+            # Check if this is a uniqueness validation error, otherwise re-raise
+            if hasattr(e, 'message_dict'):
+                for field, messages in e.message_dict.items():
+                    if any('already exists' in str(msg).lower() for msg in messages):
+                        raise DuplicateError("Medicine", [field], f"Medicine with this {field} already exists")
```

Ruff (0.14.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)